src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,69 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.WrongValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int LOW_AGE_LIMIT = 18;
+    private static final int LOW_LOGIN_LIMIT = 6;
+    private static final int LOW_PASS_LIMIT = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new WrongValidationException("User`s data is empty");
+        }
+        checkUserExist(user);
+        checkAge(user);
+        checkLogin(user);
+        checkPassword(user);
+        storageDao.add(user);
+        return user;
```

Comment:
You can inline return statement as your method add already return user
return storageDao.add(user);